### PR TITLE
Always show month of selected day in Planner

### PR DESCRIPTION
**Pull request by armstrongnate**
[run-nightly]

Also, dims days in other months.

refs: MBL-14061
affects: parent
release note: none

Test plan:
* Swipe around in Calendar, expanding and collapsing the month view
along the way
* Selecting a day in a different month should transition to that month
* Select January 31
* Collapse to week view
* Select February 1
* Expand to month view
* The month of February should be shown
* Days in other months should be dimmed

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/ahg5p48) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fahg5p48)
teacher: [Link](https://rebrand.ly/npotpim) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fnpotpim)
parent: [Link](https://rebrand.ly/cn852r8) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fcn852r8)

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024166&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-14061](https://instructure.atlassian.net/browse/MBL-14061)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.67% | 93.68% | -0.02%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Planner/CalendarDaysViewController.swift [hunk 1]
```diff
@@ -167,6 +166,7 @@ class CalendarDayButton: UIButton {
         self.date = date
         isToday = calendar.isDateInToday(date)
         isWeekend = calendar.isDateInWeekend(date)
+        isOtherMonth = calendar.compare(date, to: selectedDate, toGranularity: .month) != .orderedSame
```
Comment (vanwagonet) [inline Core/Core/Planner/CalendarDaysViewController.swift]: simple. love it.

File: Core/Core/Planner/CalendarViewController.swift [hunk 2]
```diff
@@ -183,13 +184,19 @@ class CalendarViewController: UIViewController {
         clearPageCache()
         if days.hasDate(date, isExpanded: isExpanded) {
             days.updateSelectedDate(date)
-        } else {
-            let isReverse = date < days.selectedDate
-            // Assumes selected date can't be more than 1 page away
-            let page = daysPageDelta(isReverse ? -1 : 1, from: days)
-            page.loadViewIfNeeded()
-            page.updateSelectedDate(date)
-            daysPageController.setViewControllers([ page ], direction: isReverse ? .reverse : .forward, animated: true)
```
Comment (vanwagonet) [inline Core/Core/Planner/CalendarViewController.swift]: this functionality is now missing. This was added to slide over when the list was swiped to a day in a new week/month.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Core/Core/Planner/CalendarViewController.swift (hunk 2 above)
Comment (armstrongnate) [reply Core/Core/Planner/CalendarViewController.swift]: Oooohh! That makes sense. I couldn't figure out how to trigger this path. Good to know. Fixed.

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024304&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by armstrongnate** [comment]
Okay @vanwagonet, by using `showDate` those cases should be handled (animations and all).

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024333&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**